### PR TITLE
[SectionIntro] Design guidelines

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Add `SectionIntro` design guidelines.

## List of notable changes:

- **added** new ComponentLayout to `SectionIntro` docs
- **added** design guidelines

## What should reviewers focus on?

- 👀 Documentation: https://primer-af875fe2c6-26139705.drafts.github.io/components/SectionIntro

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2231
- Tracking issue: https://github.com/github/primer/issues/1989

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/SectionIntro/index.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+## Anatomy
+
+![An image displaying a section intro with a title, description and a link.](https://github.com/primer/brand/assets/912236/719022fa-4694-42c0-8554-b1923c65739f)
+
+- Heading: The title of the section.
+- Description: A short description of the section.
+- Link: An optional link to expand on the content introduced.
```
Comment (jesskuo4) [inline apps/docs/content/components/SectionIntro/index.mdx:16]: On this page's title description, it says an "optional description" but here it says an "optional link." Hence I assume both are optional. We should probably keep this consistent depending on the use case of this component.

File: apps/docs/content/components/SectionIntro/index.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Usage
+
+Use the section intro to introduce a new section element in the page. Section intros uses full-width of the screen and can be used in conjunction with other components that expands the content introduced. For example, the [card](/components/Card) or the [pillar](/components/Pillar) components.
```
Comment (jesskuo4) [inline apps/docs/content/components/SectionIntro/index.mdx:20]:
```suggestion
Use the section intro to introduce a new section element in the page. The section intro uses the full-width of the screen and can be used in conjunction with other components that expand the content introduced. For example, the [card](/components/Card) or the [pillar](/components/Pillar) components.
```

File: apps/docs/content/components/SectionIntro/index.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    <img src="https://github.com/primer/brand/assets/912236/1d2b0be3-e696-4d4d-9b65-e8bced96fe04" />
+    <Caption>
+      Use center alignment if the section intro is paired with a centered image
+      or visual asset.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/fa2f7c30-dd11-4b56-82ed-dbcb76245c6e" />
+    <Caption>
+      Don't use center alignment if the section intro is pareid an image or
```
Comment (jesskuo4) [inline apps/docs/content/components/SectionIntro/index.mdx:59]:
```suggestion
      Don't use center alignment if the section intro is paired with an image or
```

File: apps/docs/content/components/SectionIntro/index.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      visual asset placed on the left.
```
Comment (jesskuo4) [inline apps/docs/content/components/SectionIntro/index.mdx:60]:
```suggestion
      visual asset aligned to the left.
```

**Review by jesskuo4 — COMMENTED** [review]
Hi! There are some small knit-picky changes, please feel free to change if you agree!

File: apps/docs/content/components/SectionIntro/index.mdx (hunk 1 above)
Comment (josepmartins) [reply apps/docs/content/components/SectionIntro/index.mdx:16]: Good point, I'll rephrase the `Description` description 😆

**Review by jesskuo4 — COMMENTED** [review]
🚀👍

**Review by mperrotti — APPROVED** [review]
I wasn't able to load the preview deployment, but the `.mdx` looks good.

**Pull request merged**